2018-12-26 15:31:12,064 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:31:12,611 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:38:33,444 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:38:40,403 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:38:51,813 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:40:11,160 ERROR: Exception on /unfollow/woyao [GET] [in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/chen/Documents/workspace/github/microblog/app/routes.py", line 123, in unfollow
    current_user.unfollow(username)
  File "/Users/chen/Documents/workspace/github/microblog/app/models.py", line 55, in unfollow
    if self.is_following(user):
  File "/Users/chen/Documents/workspace/github/microblog/app/models.py", line 51, in is_following
    return self.followed.filter(followers.c.followed_id == user.id).count() > 0
AttributeError: 'str' object has no attribute 'id'
2018-12-26 15:40:28,758 ERROR: Exception on /unfollow/woyao [GET] [in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/chen/Documents/workspace/github/microblog/app/routes.py", line 123, in unfollow
    current_user.unfollow(username)
  File "/Users/chen/Documents/workspace/github/microblog/app/models.py", line 55, in unfollow
    if self.is_following(user):
  File "/Users/chen/Documents/workspace/github/microblog/app/models.py", line 51, in is_following
    return self.followed.filter(followers.c.followed_id == user.id).count() > 0
AttributeError: 'str' object has no attribute 'id'
2018-12-26 15:41:57,233 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:41:57,779 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:48:34,929 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:48:35,547 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:48:55,667 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:48:56,206 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:49:45,662 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:49:46,212 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:54:58,769 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:54:59,449 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:56:46,715 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:56:47,332 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:57:32,654 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:57:33,203 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:59:40,065 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:59:40,637 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:59:51,853 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 15:59:52,434 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:00:01,340 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:00:01,958 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:03:29,406 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:03:38,555 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:03:46,232 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:03:52,382 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:03:52,960 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:08:45,381 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:08:45,947 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:11:22,803 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:11:24,051 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:11:24,663 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:13:01,251 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:54:50,035 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 16:55:40,527 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 17:01:45,559 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:44]
2018-12-26 19:23:09,698 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:46]
2018-12-26 19:25:38,078 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:46]
2018-12-26 19:34:05,353 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:46]
2018-12-26 19:52:16,400 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:46]
2018-12-26 21:27:43,771 INFO: Microblog startup [in /Users/chen/Documents/workspace/github/microblog/app/__init__.py:46]
